src/app/collective-rewards/metrics/context/CycleContext.tsx

```diff
@@ -34,54 +28,68 @@ type CycleProviderProps = {
 }
 
 export const CycleContextProvider: FC<CycleProviderProps> = ({ children }) => {
-  const [timestamp, setTimestamp] = useState(BigInt(DateTime.now().toUnixInteger()))
-
-  useEffect(() => {
-    const interval = setInterval(() => {
-      setTimestamp(BigInt(DateTime.now().toUnixInteger()))
-    }, AVERAGE_BLOCKTIME)
-
-    return () => clearInterval(interval) // Cleanup interval on component unmount
-  }, [])
+  const timestamp = useIntervalTimestamp()
+  const [cycleData, setCycleData] = useState<Cycle>({
+    cycleStart: DateTime.fromSeconds(0),
+    cycleDuration: Duration.fromObject({ seconds: 0 }),
+    cycleNext: DateTime.fromSeconds(0),
+    fistCycleStart: DateTime.fromSeconds(0),
+    endDistributionWindow: DateTime.fromSeconds(0),
+  })
 
   const {
     data: cycleStartAndDuration,
     isLoading: cycleStartAndDurationLoading,
     error: cycleStartAndDurationError,
-  } = useGetCycleStartAndDuration()
+  } = useReadCycleTimeKeeper({ functionName: 'getCycleStartAndDuration' }, { initialData: 0n })
 
   const [firstCycleStart, cycleDuration] = cycleStartAndDuration || []
 
   const {
     data: cycleStart,
     isLoading: cycleStartLoading,
     error: cycleStartError,
-  } = useGetCycleStart(timestamp)
+  } = useReadCycleTimeKeeper({ functionName: 'cycleStart', args: [timestamp] }, { initialData: 0n })
```

Comment:
```suggestion
  } = useReadCycleTimeKeeper({ functionName: 'cycleStart', args: [timestamp] })
```
We need to remove the initial data, if not the conditional in line 75 will not work.